core/templates/dev/head/pages/exploration_player/ProgressNavDirective.js

```diff
@@ -23,20 +23,41 @@ oppia.directive('progressNav', [
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/exploration_player/' +
         'progress_nav_directive.html'),
+      scope: {
+        submitFn: '&',
+        onClickContinueButton: '&'
+      },
       controller: [
         '$scope', '$rootScope', 'PlayerPositionService',
-        'PlayerTranscriptService',
+        'PlayerTranscriptService', 'ExplorationPlayerService',
+        'ExplorationPlayerStateService', 'CONTINUE_BUTTON_FOCUS_LABEL',
         function($scope, $rootScope, PlayerPositionService,
-          PlayerTranscriptService) {
-          $scope.currentCardIndex = PlayerPositionService.getActiveCardIndex();
-          $scope.transcriptLength = PlayerTranscriptService.getNumCards();
+          PlayerTranscriptService, ExplorationPlayerService,
+          ExplorationPlayerStateService, CONTINUE_BUTTON_FOCUS_LABEL) {
+          $scope.CONTINUE_BUTTON_FOCUS_LABEL = CONTINUE_BUTTON_FOCUS_LABEL;
+
+          var update = function() {
```

Comment:
If you're not using this anywhere else, may be better to declare as an anonymous function inline on line 57. If you don't want to do this then this needs a clearer name (update what?)

Reply:
changed to `updateActiveCardInfo`